examples/basics/visuals/graph.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+"""
+This example demonstrates how to visualise a NetworkX graph using the
+GraphVisual.
+"""
+
+import sys
+
+import numpy as np
+import networkx as nx
+
+from vispy import app, gloo, visuals
+from vispy.visuals.graphs import layouts
+from vispy.visuals.transforms import STTransform
+
+
+class Canvas(app.Canvas):
+    def __init__(self):
+        app.Canvas.__init__(self, title="Simple NetworkX Graph",
+                            keys="interactive", size=(600, 600))
+
+        self.graph = nx.fast_gnp_random_graph(1000, 0.0006, directed=True)
+
+        self.visual = visuals.GraphVisual(
+            # np.asarray(nx.to_numpy_matrix(self.graph)),
+            nx.adjacency_matrix(self.graph),
```

Comment:
Right, that's what it is -- and the matrix operates differently from an array with the `*` operator, see my comment:

https://github.com/vispy/vispy/pull/1043#issuecomment-130733060

So the code should probably ensure that, if it's a matrix, it gets cast to an array so that this error doesn't happen.